Support Charset.aliases() method

cquiroz opened the pull request:
This is a small addition to `Charset` to support the method `aliases()` basically exposing the `aliases` field in the right format.

This is the method in particular:
https://docs.oracle.com/javase/7/docs/api/java/nio/charset/Charset.html#aliases()

scala-jenkins commented:
Can one of the admins verify this patch?

sjrd commented on javalib/src/main/scala/java/nio/charset/Charset.scala:
```diff
@@ -8,6 +8,9 @@ abstract class Charset protected (canonicalName: String,
     aliases: Array[String]) extends AnyRef with Comparable[Charset] {
```
Could you rename this to `_aliases` or `aliasesArray`, so that there isn't a semi-clash between the private `aliases` and the `def aliases()`?

sjrd commented on javalib/src/main/scala/java/nio/charset/Charset.scala:
```diff
@@ -8,6 +8,9 @@ abstract class Charset protected (canonicalName: String,
   final def name(): String = canonicalName
 
+  final def aliases(): java.util.Set[String] =
+    new java.util.HashSet(java.util.Arrays.asList(aliases))
```
The Javadoc says that `aliases()` must return an *immutable* set of the aliases. Therefore, you should wrap the produced `HashSet` with [`java.util.Collections.unmodifiableSet`](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#unmodifiableSet-java.util.Set-). This also means that you can always reuse the same unmodifiable set, which you can store in a `private lazy val aliasesSet`. Make sure to make it lazy so that all this stuff can be dce'ed away if `aliases()` is not called.

sjrd commented on test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/CharsetTest.scala:
```diff
@@ -67,4 +69,24 @@ class CharsetTest {
+
+  @Test def aliases(): Unit = {
+    assertEquals(Charset.forName("UTF-8").aliases(),
+      List("UTF8", "unicode-1-1-utf-8").toSet.asJava)
```
Use `Set(...)` instead of `List(...).toSet`. Even better, define a local
```scala
def javaSet[A](elems: A*): java.util.Set[A] = Set(elems: _*).asJava
```
and then use `javaSet("UTF8", "unicode-1-1-utf-8")`.

sjrd replied:
Also, we use 4 spaces indentation when continuing a line. See our coding style at https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#indentation

cquiroz commented:
Thanks for your comments. I rebased to 0.6.x, addressed your concerns and squashed
Please let me know if you need more changes

How are these commits added to master?

sjrd commented:
Dear Jenkins, test this please

sjrd commented on test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/CharsetTest.scala:
```diff
@@ -67,4 +70,24 @@ class CharsetTest {
+
+  @Test def aliases(): Unit = {
+    assertEquals(Charset.forName("UTF-8").aliases(),
+      javaSet("UTF8", "unicode-1-1-utf-8"))
```
This is in fact also the continuation of the line starting with `assertEquals`, so this should be indented 4 spaces.

sjrd commented on test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/CharsetTest.scala:
```diff
@@ -67,4 +70,24 @@ class CharsetTest {
+      javaSet("UTF_16", "unicode", "utf16", "UnicodeBig"))
+    assertEquals(Charset.forName("UTF-16BE").aliases(),
+      javaSet("X-UTF-16BE", "UTF_16BE", "ISO-10646-UCS-2",
+          "UnicodeBigUnmarked"))
```
Here, you'll have 4 spaces of indentation on top of 4 spaces of indentation, so at the end of the day those lines should look like:
```scala
    assertEquals(Charset.forName("UTF-16BE").aliases(),
        javaSet("X-UTF-16BE", "UTF_16BE", "ISO-10646-UCS-2",
            "UnicodeBigUnmarked"))
```

cquiroz commented:
Ok, I hope I got it right this time 😄 
Changes were squashed

scala-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 

<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4439/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4439/</a>
<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5993/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5993/</a>
Test PASSed.

sjrd commented:
Dear Jenkins, retest this please

scala-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 

<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4441/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4441/</a>
<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5995/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5995/</a>
Test PASSed.
